resources/main_test.go

```diff
@@ -0,0 +1,132 @@
+package resources
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/terraform"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+var debugFlag = false
+
+func TestMain(m *testing.M) {
+	// call flag.Parse() here if TestMain uses flags
+
+	// Setup
+	os.Setenv("TF_IN_AUTOMATION", "1")
+	if err := RunCmd("terraform", "init", "-input=false", "-no-color"); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Test
+	code := m.Run()
+
+	// Teardown
+	fmt.Println("Teardown: destroying all terraform resources of the test suite")
+	RunCmd("terraform", "destroy", "-input=false", "-no-color", "-auto-approve")
+
+	os.Exit(code)
+}
+
+func ResourceTypeTest(tfType, tfName, nukeType string, t *testing.T) error {
```

Comment:
`*testing.T` should be the first parameter, because this is sort of a uber-object, which gets passed into all test helpers.

Reply:
Since you have `*testing.T` you don't need to return an `error`. You could fail directly.